### PR TITLE
Add beekeeping

### PR DESCRIPTION
**Pull request by Rail-Runner**
#### Summary

SUMMARY: Content "Add beekeeping"
#### Purpose of change

Allow players to build beehives for a renewable source of honey and wax.
#### Describe the solution

Add JSON definitions for placed beehives, empty beehives in item from, construction recipe to place the beehive, and crafting recipe for empty beehive.
#### Describe alternatives you've considered

Making use of "bees" field in some way, though it feels rather broken right now, and also seems like some beekeepers are able to work with their hives without protective gear and without being stung (or at least stung enough that it'd have major in-game effects).
#### Testing

Copy-pasted the changes to game files, made sure that there are no errors on startup.
#### Additional context

Taken from MST Extra mod, with some changes made after looking up some info about beekeeping.

### CODE REVIEW

File: data/json/construction.json [hunk 1]
```diff
@@ -71,6 +71,21 @@
...
+    "type": "construction",
+    "id": "constr_beehive",
+    "group": "build_beehive",
+    "category": "CONSTRUCT",
+    "required_skills": [ [ "survival", 4 ] ],
+    "time": "30 m",
+    "qualities": [ [ { "id": "DIG", "level": 1 } ] ],
+    "components": [
+      [ [ "beehive_empty", 1 ] ],
+      [ [ "wax", 1 ], [ "sugar", 1 ], [ "sweet_water", 1 ], [ "molasses", 1 ], [ "syrup", 1 ], [ "beet_syrup", 1 ] ]
```
[the remark this answers is not in the document]
Comment (Rail-Runner) [reply data/json/construction.json:84]: Not equivalent, but it is one of things that can be used to lure bees into the beehive, from what I read. That's what sugary things are used for here.

[the remark this answers is not in the document]
Comment (Fris0uman) [reply data/json/construction.json:84]: Ah right, but there's already wax in the empty beehive recipe

**Comment by Fris0uman** [comment]
So if I put that in my basement i'll have honey forever for free if I just wait long enough?

File: data/json/construction.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rail-Runner) [reply data/json/construction.json:84]: After some more reading, seems like wax is only necessary for top-bar hives (while I assumed that this hive is a frame hive, due to being of the most simple construction). Guess I'll remove wax from empty beehive recipe.

**Comment by anothersimulacrum** [comment]
This looks quite hacky. While I'd love to have bees, I'd like to have them properly implemented.

**Comment by Rail-Runner** [comment]
> So if I put that in my basement i'll have honey forever for free if I just wait long enough?

Nope, beehives should be placed on a diggable tile (like soil), which would need to be outside. I guess if basements can have such tiles of diggable soil, you could also grow plants there as well as place beehives, so they probably shouldn't have that.

**Comment by Venera3** [comment]
Bees are cool, and beekeeping is definitely something I'd like to be able to do ( and in the game..). That said:

Is there a way to restrict the filling of an empty hive to summer, since this implementation simulates the capture of a wild swarm? (definitely not essential if it's not possible)
How long does a (empty/harvested) hive take to produce? Generally hives are harvested once or twice a year, and that's ignoring the first year when a proper harvest isn't even guaranteed. 
Also I don't think survival scaling is appropriate, I'd rather go with a broader range of results since honey yields can be very variable but aren't that skill-based.

**Comment by Positronic-Girl** [comment]
I agree with simulacrum, my experiences with MST extra bees, that while they do the job, they are very hacky, which is a a product of it being a json mod, if we are mainlining bees, I'd much prefer if they were implemented in a much better method with C++ support, instead of json hacks

**Comment by Rail-Runner** [comment]
> Is there a way to restrict the filling of an empty hive to summer, since this implementation simulates the capture of a wild swarm? (definitely not essential if it's not possible)

No way to do that, as far as I know.

> How long does a (empty/harvested) hive take to produce? Generally hives are harvested once or twice a year, and that's ignoring the first year when a proper harvest isn't even guaranteed.

Should be the the same as fruit trees currently do; probably once per year; except it takes twice that time (two years) before you can harvest products for the first time.

**Comment by Fris0uman** [comment]
Accidentally merged before being ready, feel free to reopen

**Comment by I-am-Erk** [comment]
Apologies for the false start there. This thread will need to be re created if you wish to pursue this feature.

**Comment by themaster567** [comment]
*drops prepared dinner in the floor*
Whoops, you'll need to make it again if you want me to eat it. Nice.

**Comment by I-am-Erk** [comment]
Yep, it's lame.

**Pull request merged**